inventory_management_system_api/services/utils.py

```diff
@@ -142,8 +142,9 @@ def _add_catalogue_item_property_units(
     :param supplied_properties: The supplied catalogue item properties.
     """
     logger.info("Adding the units to the supplied properties")
-    for supplied_property_id, supplied_property in supplied_properties.items():
-        supplied_property["unit"] = defined_properties[supplied_property_id]["unit"]
+    for supplied_property_name, supplied_property in supplied_properties.items():
+        supplied_property["unit"] = defined_properties[supplied_property_name]["unit"]
+        supplied_property["unit_id"] = defined_properties[supplied_property_name]["unit_id"]
```

Comment:
```suggestion
        supplied_property["unit_id"] = defined_properties[supplied_property_name]["unit_id"]
        supplied_property["unit"] = defined_properties[supplied_property_name]["unit"]
```
Just for consistency.